authui/src/select.ts

```diff
@@ -20,18 +20,17 @@ export class SelectEmptyValueController extends Controller {
   }
 }
 
-interface GenderSelect {
-  selectElement: HTMLSelectElement;
-  inputElement: HTMLInputElement;
-  listener: (e: Event) => void;
-}
+export class GenderSelectController extends Controller {
+  static targets = ["select", "input"];
+
+  declare selectTarget: HTMLSelectElement;
+  declare inputTarget: HTMLInputElement;
+
+  toggle({ params }: { params?: { fromlistener: boolean } }) {
```

Comment:
This style is no good. `this.toggle({})` looks strange to me.
`this.toggle` sometimes take a `Event` as the first argument, and sometimes take an empty object.

```typescript
onChange() {
  this.toggle(true);
}

connect() {
  this.toggle(false);
}

toggle(fromListener: boolean) {
  // ...
}
```